OsuPlayer.Network/API/Service/AbstractApiBase.cs

```diff
@@ -34,6 +35,10 @@ protected void ParseWebException(Exception ex)
     /// </summary>
     protected void CancelCancellationToken()
     {
+        // TODO: Currently does nothing, because it breaks the functionality a bit. 
+        // TODO: This needs to be re-thinked a bit!
+        return;
+        
         CancellationTokenSource.Cancel();
         CancellationTokenSource = new ();
```

Comment:
Commenting these lines out would be better as this code is currently not reachable. The TODO is explaining accordingly.